### PR TITLE
Tests to verify that the failover and relocate operations are not affected by node drain

### PR DESCRIPTION
**Pull request by jilju**
Tests cases to verify that the failover and relocate operations are not affected when node is drained while the failover/relocate operations are going on.

<google-sheets-html-origin>

OCS-4441 | ([RHSTOR-2139](https://issues.redhat.com//browse/RHSTOR-2139) Regional DR for RBD - Mature from TP to GA) During Failover from C1 to C2, Drain the node in C2 where the rbd mirror pod is running
-- | --
OCS-4442 | ([RHSTOR-2139](https://issues.redhat.com//browse/RHSTOR-2139) Regional DR for RBD - Mature from TP to GA) During Relocate from C2 to C1, Drain the node in C1 where the rbd mirror pod is running
OCS-4443 | ([RHSTOR-2139](https://issues.redhat.com//browse/RHSTOR-2139) Regional DR for RBD - Mature from TP to GA) During Failover from C1 to C2, Drain the node in C2 where ODR operator pod is running
OCS-4444 | ([RHSTOR-2139](https://issues.redhat.com//browse/RHSTOR-2139) Regional DR for RBD - Mature from TP to GA) During Relocate from C2 to C1, Drain the node in C1 where ODR operator pod is running

### CODE REVIEW

**Comment by jilju** [comment]
Subscription workload tests verified: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/jijoy-rdr-a/jijoy-rdr-a_20231102T094024/multicluster/logs/test_report_1699331730.html

File: tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                *[constants.SUBSCRIPTION, "ramen_dr_cluster_operator"],
+                marks=pytest.mark.polarion_id("OCS-4443"),
+            ),
+            pytest.param(
+                *[constants.APPLICATION_SET, "rbd_mirror"],
+                marks=pytest.mark.polarion_id("OCS-4442"),
+            ),
+            pytest.param(
+                *[constants.APPLICATION_SET, "ramen_dr_cluster_operator"],
+                marks=pytest.mark.polarion_id("OCS-4444"),
```
[the remark this answers is not in the document]
Comment (jilju) [reply tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py:47]: Yes, that is correct.

File: tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    drain_nodes,
+    schedule_nodes,
+)
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running, get_pods_having_label
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+@tier4b
```
[the remark this answers is not in the document]
Comment (jilju) [reply tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py:23]: Added squad mark

File: tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jilju) [reply tests/disaster-recovery/regional-dr/test_node_operations_during_failover_relocate.py:48]: okay

**Review by keesturam — APPROVED** [review]
LGTM

**Pull request merged**